lifecycle.go

```diff
@@ -0,0 +1,175 @@
+package main
```

Comment:
lifecycle.go will be the entry point(with main func) in a container. Each container will have only one lifecycle.go. lifecycle.go is supposed to manage that container's lifecycle(i.e. when to stop spawning, etc)